iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/LongRunInstructionCheck.java

```diff
@@ -118,11 +130,47 @@ private static void incrementWordCountOnLine(Map<Integer, Integer> wordsPerLine,
   }
 
   private static void incrementWordCountOnLine(Map<Integer, Integer> wordsPerLine, int line) {
-    wordsPerLine.putIfAbsent(line, 0);
-    wordsPerLine.compute(line, (key, value) -> value + 1);
+    wordsPerLine.merge(line, 1, Integer::sum);
+  }
+
+  private static Set<Integer> getLinesWithUrl(RunInstruction runInstruction) {
+    return runInstruction.arguments().stream()
+      .filter(LongRunInstructionCheck::isArgumentUrl)
+      .flatMap((Argument urlArgument) -> {
+        int startLine = urlArgument.textRange().start().line();
+        int endLine = urlArgument.textRange().end().line();
+        return IntStream.rangeClosed(startLine, endLine).boxed();
+      })
+      .collect(Collectors.toSet());
+  }
+
+  private static boolean isArgumentUrl(Argument argument) {
+    return argument.expressions().stream().anyMatch(LongRunInstructionCheck::isExpressionUrl);
+  }
+
+  private static boolean isExpressionUrl(Expression expression) {
+    return switch (expression.getKind()) {
+      case STRING_LITERAL -> isStringUrl(((Literal) expression).value());
+      case EXPANDABLE_STRING_LITERAL -> ((ExpandableStringLiteral) expression).expressions().stream().anyMatch(LongRunInstructionCheck::isExpressionUrl);
+      case EXPANDABLE_STRING_CHARACTERS -> isStringUrl(((ExpandableStringCharacters) expression).value());
+      default -> false;
+    };
+  }
+
+  private static boolean isStringUrl(String value) {
+    try {
+      new URL(value).toURI();
+      return true;
+    } catch (MalformedURLException | URISyntaxException e) {
+      return false;
+    }
```

Comment:
```suggestion
    Pattern regex = Pattern.compile("\\b(?:(https?|ftp|file)://|www\\.)?[-A-Z0-9+&#/%?=~_|$!:,.;]*[A-Z0-9+&@#/%=~_|$]\\.[-A-Z0-9+&@#/%?=~_|$!:,.;]*[A-Z0-9+&@#/%=~_|$]", Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE);
    Matcher regexMatcher = regex.matcher(value);
    return regexMatcher.matches();
```

Can use regex for validation and handle URLs without protocol, unsure if it's better or not.

Reply:
I think it's safe to assume that we will see full-qualified URLs in Dockerfiles more often than not. For example, in test resources in sonar-iac we have no examples of `www.` being used without a protocol in Dockerfiles.